dataclasses_json/core.py

```diff
@@ -41,7 +41,7 @@ def _decode_dataclass(cls, kvs, infer_missing):
                                                         field_value,
                                                         infer_missing)
 
-        elif _is_supported_generic(field.type) and field.type != str:
+        elif _is_supported_generic(field.type) and field.type != str and field.type.__args__[0] != str:
```

Comment:
Thanks for putting in this fix, but I've found the issue actually stems from `str` passing the `_is_supported_generic` check. See #16